npeps/npep-308.md

```diff
@@ -0,0 +1,127 @@
+clusters.
+
+## Goals
+
+* To establish a standard, agnostic convention for multi-cluster network
+  policies within the existing `NetworkPolicy` resource.
+* To enhance portability of network security policies across various
+  multi-cluster Kubernetes environments.
+* To avoid the proliferation of custom CRDs for multi-cluster policy definition,
+  hence simplifying the user experience.
+
+## Non-Goals
+
+* This proposal does **not** define the architecture or implementation of the
+  multi-cluster control or data planes.
+* It does **not** specify how projects should implement the underlying network
+  connectivity and policy enforcement between clusters.
+
+## Introduction
+
+The current Kubernetes `NetworkPolicy` API is limited to intra-cluster traffic.
+As multi-cluster architectures become more common, there is a growing need for a
+standardized way to manage traffic flow between services in different clusters.
+Today, this is handled by project-specific solutions, such as Cilium's
+`CiliumNetworkPolicy` or Calico's `GlobalNetworkPolicy`. This leads to a
+fragmented ecosystem where security policies are not portable and useßrs are
+locked into a specific project for multi-cluster functionality.
+
+This proposal aims to address this by introducing a simple, backward-compatible
+extension to the `NetworkPolicy` API that will provide a unified way to define
+cross-cluster communication rules.
+
+## User-Stories/Use-Cases
+
+Story 1: Secure Applications Communication Across Clusters
+
+* **As a** DevOps engineer,
+* **I want to** define a `NetworkPolicy` that allows a `frontend` service in
+  `cluster-1` to communicate with a `backend` service in `cluster-2`,
+* **so that** I can enforce fine-grained security policies for my application,
+  regardless of which cluster the services are running in.
+
+Story 2: Isolate Development and Production Environments
+
+* **As a** platform administrator,
+* **I want to** create a default-deny policy for a namespace in my development
+  cluster that allows egress traffic only to specific services in my production
+  cluster,
+* **so that** I can prevent unauthorized access from the development environment
+  to sensitive production resources.
+
+## API
+
+We propose the introduction of a new reserved label,
+`networking.k8s.io/multicluster-name`, to be used within the `podSelector` and
+`namespaceSelector` of `NetworkPolicy` resources. When this label is present,
+the CNI is responsible for interpreting it as a selector for resources in a
+remote cluster.
+
+**Importantly, the addition of this label will have no impact on CNIs or other
+projects that do not implement this functionality.** If a CNI does not recognize
+the `networking.k8s.io/multicluster-name` label, it will simply be ignored, and
+the `NetworkPolicy` will be enforced as if the label was not present. This
+ensures full backward compatibility and means that this change will not break
+any existing `NetworkPolicy` implementations.
```

Comment:
It won't be "ignored", it will be interpreted as being a local label. I guess we assume that nobody will _actually_ label namespaces with this label.